src/R8R.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+
+contract R8R is Ownable, ReentrancyGuard {
+    // =========================
+    // === STORAGE VARIABLES ===
+    // =========================
+
+    address public robot;
+    uint256 public gameId;
+    uint256 public gameEntryPriceInEth;
+
+    mapping(uint256 gameId => Game) games;
+
+    IERC20[] public gameTokensAllowedList; // list of ERC20 tokens users can use to enter the game
+    uint256[] public gameTokensEntryPriceList; // list of entry price for the ERC20 tokens
+
+    struct Game {
+        uint256 gameId;
+        address[] players;
+        uint256[] playerRatings;
+        uint256 robotRating;
+        address winner;
+        uint256 winnerPayout;
+    }
+
+    // ==============
+    // === EVENTS ===
+    // ==============
+
+    event GameCreated(uint256 gameId);
```

Comment:
Events are going to be our greatest source of truth for ponder, so we want these to be as verbose as possible. What other variables would be good to emit when the game is created? Think about what the frontend needs to know to render a game after it has been created on-chain.